frontend/src/components/PageContent.js

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import Section from "./Section";
+
+class PageContent extends React.Component {
+  constructor(props) {
+      super(props);
+
+      this.state = {error: null, sections: []};
+  }
+
+  componentDidMount() {
+    this.updatePageFromServer()
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.wikiId !== prevProps.wikiId || this.props.pageId !== prevProps.pageId) {
+      this.updatePageFromServer()
+    }
+  }
+
+  updatePageFromServer() {
+    if (this.props.wikiId === null || this.props.pageId === null) return
+
+    fetch(`http://localhost:8000/api/w/${this.props.wikiId}/p/${this.props.pageId}/s`, { crossDomain: true })
+      .then((res) => res.json())
+      .then(
+        (returnedSections) => {
+          console.log(returnedSections)
+            if (Array.isArray(returnedSections)) {
+              this.setState({sections: returnedSections});
```

Comment:
Do you want to set `error` to `null` in this case? And in the case below, do you want to set `sections` to `[]`?

Reply:
Excellent point, addressed in a26741fa7ee24